scripts/zfs-tests.sh

```diff
@@ -695,17 +695,21 @@ fi
 #
 # Run all the tests as specified.
 #
+EVILRES=$(mktemp -u -t test-runner-error)
 msg "${TEST_RUNNER} ${QUIET:+-q}" \
     "-c \"${RUNFILES}\"" \
     "-T \"${TAGS}\"" \
     "-i \"${STF_SUITE}\"" \
     "-I \"${ITERATIONS}\""
-${TEST_RUNNER} ${QUIET:+-q} \
+(${TEST_RUNNER} ${QUIET:+-q} \
     -c "${RUNFILES}" \
     -T "${TAGS}" \
     -i "${STF_SUITE}" \
     -I "${ITERATIONS}" \
-    2>&1 | tee "$RESULTS_FILE"
+    2>&1 ; echo $? > "$EVILRES"; )| tee "$RESULTS_FILE"
+
+RUNRESULT=$(cat "$EVILRES")
```

Comment:
```sh
{ ${T_R}
    -...
2>&1; RUNRESULT=$?; } | tee "$R_F"
```
and you can drop the temp file

Reply:
Tragically...
zfs-tests.sh:708:12: note: Modification of RUNRESULT is local (to subshell caused by pipeline). [SC2030]
zfs-tests.sh:753:7: note: RUNRESULT was modified in a subshell. That change might be lost. [SC2031]

So should I silence it, do you think?

Reply:
> ```shell
> { ${T_R}
>     -...
>     2>&1; RUNRESULT=$?; } | tee "$R_F"
> ```
The snippet above still spawns a subshell for the pipeline.

Reply:
I'd be more than happy to hear less gross alternatives than stashing the return value in a file and retrieving it, I just don't currently see any.

Reply:
Yeah, the lint's right, I overlooked that. I don't think there's a better way than doing it this way with /bin/sh.

Reply:
I think using tempfile is fine. I don't see any other obvious way.
Lesson learned from this issue: we shouldn't overlook having `set -e` at the beginning of every shell script :)